core/src/main/java/org/apache/iceberg/view/ViewPropertiesUpdate.java

```diff
@@ -0,0 +1,127 @@
+
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT;
+
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.util.Tasks;
+
+public class ViewPropertiesUpdate implements UpdateViewProperties {
+  private final ViewOperations ops;
+  private final Map<String, String> updates = Maps.newHashMap();
+  private final Set<String> removals = Sets.newHashSet();
+  private ViewMetadata base;
+
+  ViewPropertiesUpdate(ViewOperations ops) {
+    this.ops = ops;
+    this.base = ops.current();
+  }
+
+  @Override
+  public Map<String, String> apply() {
+    this.base = ops.refresh();
+
+    Map<String, String> newProperties = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : base.properties().entrySet()) {
+      if (!removals.contains(entry.getKey())) {
+        newProperties.put(entry.getKey(), entry.getValue());
+      }
+    }
+
+    newProperties.putAll(updates);
+    return newProperties;
+  }
+
+  @Override
+  public void commit() {
+    Tasks.foreach(ops)
+        .retry(
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_NUM_RETRIES, COMMIT_NUM_RETRIES_DEFAULT))
+        .exponentialBackoff(
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_MIN_RETRY_WAIT_MS, COMMIT_MIN_RETRY_WAIT_MS_DEFAULT),
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_MAX_RETRY_WAIT_MS, COMMIT_MAX_RETRY_WAIT_MS_DEFAULT),
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_TOTAL_RETRY_TIME_MS, COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT),
+            2.0 /* exponential */)
+        .onlyRetryOn(CommitFailedException.class)
+        .run(
+            taskOps -> {
+              Map<String, String> newProperties = apply();
+              long timestampMillis = System.currentTimeMillis();
+              int newVersion = base.currentVersionId() + 1;
+              ViewMetadata updated =
+                  ImmutableViewMetadata.builder()
+                      .from(base)
+                      .currentVersionId(newVersion)
```

Comment:
This should not be modified by a property update. Changing properties does not alter the current version of the view. I think the confusion is that there is a table "version" that is updated every time we write the metadata file. But that version is not the format version nor the snapshot ID, it's just a counter that we increment to help identify the metadata files.

This version is more like the snapshot ID, but it is assigned sequentially. This should be entirely handled by the `ViewOperations`, if we keep that interface.

Reply:
I wasn't entirely sure at the beginning but in hindsight I agree that changing properties of a view shouldn't change the view's version id. I've updated the code and the test to reflect this